### PR TITLE
Add methods to get group/server/user icon URLs

### PR DESCRIPTION
**Pull request by ghost**
Due to the avatar/icon string provided by the api being only a hash,
there is no user-friendly way to get the direct URLs for groups,
servers, and users' avatars/icons. This will allow retrieving these URLs
easier.

### CODE REVIEW

File: src/model.rs [hunk 1]
```diff
@@ -490,6 +514,14 @@ impl Group {
 			}
 		}
 	}
+
+	/// Returns the formatted URL of the group's icon.
+	///
+	/// Returns None if the group does not have an icon.
+	pub fn icon_url(&self) -> Option<String> {
+		self.icon.as_ref().and_then(|icon|
+			Some(format!(cdn_concat!("channel-icons/{}/{}.jpg"), self.channel_id, icon)))
```
Comment (SpaceManiac) [inline src/model.rs]: Missing leading `/`, I think.

File: src/model.rs [hunk 2]
```diff
@@ -248,6 +248,14 @@ impl ServerInfo {
 			permissions: try!(remove(&mut value, "permissions").and_then(Permissions::decode)),
 		})
 	}
+
+	/// Returns the formatted URL of the server's icon.
+	///
+	/// Returns None if the server does not have an icon.
+	pub fn icon_url(&self) -> Option<String> {
+		self.icon.as_ref().and_then(|icon|
+			Some(format!(api_concat!("/guilds/{}/icons/{}.jpg"), self.id, icon)))
```
Comment (SpaceManiac) [inline src/model.rs]: No need to write `.and_then(|_| Some(...))` - using `.map(|_| ...)` should be fine.

**Comment by ghost** [comment]
Updated.

(of course groups is the _one_ I don't test and it was wrong)

File: src/lib.rs [hunk 3]
```diff
@@ -47,6 +47,16 @@ mod connection;
 mod state;
 #[cfg(feature="voice")]
 pub mod voice;
+
+#[macro_use]
```
Comment (SpaceManiac) [inline src/lib.rs]: These two `#[macro_use]` annotations are unnecessary.

**Comment by SpaceManiac** [comment]
Do you know whether it is currently preferred to hit the API or the CDN directly for these?

**Comment by ghost** [comment]
Updated.

It looks like it's actually neither **discordapp.com/api** nor **cdn.discordapp.com**, it's **discordcdn.com**, which they've been switching to recently.

**Comment by ghost** [comment]
Updated to use **cdn.discordapp.com**, per:

```
zey - Today at 12:30 PM
jake
quick question
when generating URLs to icons/avatars is it preferred to use cdn.discordapp.com or discordcdn.com
⎛⎝xX_jake_Xx⎠⎞ - Today at 12:30 PM
its preferred to use cdn.discordapp.com
zey - Today at 12:30 PM
okay thanks
⎛⎝xX_jake_Xx⎠⎞ - Today at 12:30 PM
probably gonna throw away discordcdn.com
not sure yet
```

**Pull request merged**